packages/shared/src/encryption/signature.ts

```diff
@@ -0,0 +1,115 @@
+import crypto from "crypto";
```

Comment:
I assume this is copied and well tested? On skim this looks reasonable with no red flags going up.